pkg/certs/cert-inspection/certgraphanalysis/metadata_options.go

```diff
@@ -44,3 +105,48 @@ var (
 		},
 	}
 )
+
+func RewriteNodeIPs(nodeList *corev1.NodeList) *metadataOptions {
+	nodes := map[string]int{}
+	for i, node := range nodeList.Items {
+		nodes[node.Name] = i
+	}
+	return &metadataOptions{
+		rewriteSecret: func(secret *corev1.Secret) {
+			rewriteIPInEtcdServingMetricsSecret(secret, nodes)
+			rewriteIPInEtcdServingSecret(secret, nodes)
+			rewriteIPInEtcdPeerSecret(secret, nodes)
+		},
+	}
+}
+
+func rewriteIPInEtcdServingSecret(secret *corev1.Secret, nodes map[string]int) {
+	if secret.Namespace == "openshift-etcd" && strings.HasPrefix(secret.Name, "etcd-serving-") {
```

Comment:
why bound to etcd?  We want to do this for *all* secrets, right?  So the thing you're trying to replace is the master name.

Reply:
Same question throughout.